sync/class.jetpack-sync-module-meta.php

```diff
@@ -70,9 +70,10 @@ public function get_objects_by_id( $object_type, $config ) {
 	}
 
 	public function init_listeners( $callable ) {
-		$whitelist_handler = array( $this, 'filter_meta' );
+
 
 		foreach ( $this->meta_types as $meta_type ) {
+			$whitelist_handler = array( $this, 'filter_meta_' . $meta_type );
```

Comment:
I would move this line above `add_filter( "jetpack_sync_before_enqueue_added_{$meta_type}_meta", $whitelist_handler );`